### ISSUE TITLE
easy window resizing and template positioning

### ISSUE THREAD

**Issue by commandline-be** [hyprwm/Hyprland#7960]
### Description

New to hyprland and happy to see a reasonably easy configurable WM growing popularity.

The window placement is mostly pleasant to work with.

* Being able to drag the size of a window looks like a welcome feature, this would not change positioning, just size

* the ability to configure a template for window positioning could prove practical, for example like 

    Master (left, 70%) Master (left, 30%) Other (right, 60%) Other (right, 40%)

* the ability to assign such templates to workspaces and/or multiple master-layout profiles also seems interesting

**Comment by fufexan** [comment]
The first point is addressed by the option `general:resize_on_border = true`. For the rest, a layout plugin might be more suitable (unless there are already scripts that do what you want).

**Comment by commandline-be** [comment]
> The first point is addressed by the option `general:resize_on_border = true`. For the rest, a layout plugin might be more suitable (unless there are already scripts that do what you want).

Thanks for that, I must have overlooked this.

If a script can do that I may have a go at it and turn it into a plugin.

**Comment by commandline-be** [comment]
also my compliments on a very well made product, thank you
in the early days I used blackbox and fluxbox, later metacity
somehow I never grew the same comfort with modern desktops untile with hyprland

**Comment by vaxerski** [comment]
I don't understand point 2 and 3

**Comment by commandline-be** [comment]
> I don't understand point 2 and 3

Attempt at splitting up that my understanding it is one (master, dwindle) policy applied to all workspaces, looking to have a policy associated with a worksapce and another

A template could look like setting what app to open in what window at what size of window

**Comment by vaxerski** [comment]
what?

**Comment by commandline-be** [comment]
> what?

please elaborate

**Comment by vaxerski** [comment]
I just don't understand what you want to achieve at all

**Comment by commandline-be** [comment]
basically, to set multiple templates for how application windows can be spread in a workspace

say i call a template 'code' which offers one big window for code at 60% width next to that one at 60% for reading a PDF, one at 40% for keeping a terminal open. to go as far as to create a key-binding to load such a template with pre-configured applications to open at the workspace of choice

from what i gathered this is not explicitely possible but could be made possible

**Comment by commandline-be** [comment]
> The first point is addressed by the option `general:resize_on_border = true`. For the rest, a layout plugin might be more suitable (unless there are already scripts that do what you want).

cannot believe I overled that the resize_on_border was actually in the template config, works great

**Comment by vaxerski** [comment]
yeah you can select which workspace apps open on (rules) but setting templates isnt possible

**Comment by kRHYME7** [comment]
This is what I understood 
(sample)
![image](https://github.com/user-attachments/assets/60501dcf-ae28-4256-8d04-9ab80b515f1c)

This surely is a good feature, but I do not have any idea if this is possible for  the 2 layouts dwindle and master to have 

For dwindle I can think of a workaround using 
` default_split_ratio `  
![image](https://github.com/user-attachments/assets/938f09ce-bb16-43dc-be89-3150a1441345)

One can write a program to launch apps respectively with their own ratio.

apps A,B, and C can have their own ratio (by pair) so 

example 
A+B can have 0.9  then C will depend on B.


Templating is certainly a good feature to have.

**Comment by commandline-be** [comment]
> This is what I understood (sample) ![image](https://private-user-images.githubusercontent.com/53417443/372771495-60501dcf-ae28-4256-8d04-9ab80b515f1c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4NzQyOTEsIm5iZiI6MTcyNzg3Mzk5MSwicGF0aCI6Ii81MzQxNzQ0My8zNzI3NzE0OTUtNjA1MDFkY2YtYWUyOC00MjU2LThkMDQtOWFiODBiNTE1ZjFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAyVDEyNTk1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxYzBjNGY5OTljNDBjMmRhNjFiOWZjOGIzNDJmNjkxY2I5MWVjOWViMzhkOTRlY2UzMDM1MDdlNTYzNDkzMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.yzDY6N7qOozT97ZO8BLLdIT55p3E3GNFHIjXakO_Y2g)
> 
> This surely is a good feature, but I do not have any idea if this is possible for the 2 layouts dwindle and master to have
> 
> For dwindle I can think of a workaround using `default_split_ratio` ![image](https://private-user-images.githubusercontent.com/53417443/372774599-938f09ce-bb16-43dc-be89-3150a1441345.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4NzQyOTEsIm5iZiI6MTcyNzg3Mzk5MSwicGF0aCI6Ii81MzQxNzQ0My8zNzI3NzQ1OTktOTM4ZjA5Y2UtYmIxNi00M2RjLWJlODktMzE1MGExNDQxMzQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAyVDEyNTk1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhYWE4NTk1MWFiZGJmYzlhZTYwNGU1MTNiMzliZjg1OGQyZGI5ODNjZDIzM2FiZjBlODM3ZDhkYmVlMmY3NzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HjEb_glsv4XN5Vik7L45D1v35ZFQ_2J9MWB8P3nOkg8)
> 
> One can write a program to launch apps respectively with their own ratio.
> 
> apps A,B, and C can have their own ratio (by pair) so
> 
> example A+B can have 0.9 then C will depend on B.
> 
> Templating is certainly a good feature to have.

very much on point with what i have in mind

**Comment by commandline-be** [comment]
I don't think the configuration for templates should be too extensive.

It could be that such is a third option next to master and dwindle.
sharing the below to consider options, it may be too convoluted

To asume for a moment layout engine = templated
---

workspace = main, M:L(F,40%, $myapp1), S:(30%,,$myapp2),S:(R,,$myapp3),S:(20%,,$myapp4)
workspace = dev, M:C(F,50%,$myapp6),S:L(10%,,$myapp7),S:L

syntax be like

<name>,
M=master, S=Slave : L(eft) R(ight) C(enter) T(op) B(ottom) M(iddle)  (<height>,<width>,<exec>)
where inside the () F(ull) R(emainder) E(qualise)

---

something like that

**Comment by enk-it** [comment]
So would it have fixed number of pre-configured windows? What happens if one opens more than his pre-configured layout has? Looks like it could be new option (master, dwindle, static?) but i think it's a lot of work.

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
